File: web-development-101/test-driven-development/javascript-exercises/pig_latin/pigLatin.js
function translate(text) {
	words = text.split(' ');
	newWordsArray = [];
	for (let i = 0; i < words.length; ++i) {
		newWordsArray.push(getPigLatin(words[i]))
	}
	return newWordsArray.join(' ');
}

function getPigLatin(word) {
	let vowelPattern = /^[aeiou]+/i;
	let consonantPattern = /^[^aeiou]+/i;
	let quPattern = /[^aeiou]*qu/i;
	if(word.match(quPattern)) {
		let splitIndex = quPattern.exec(word).index +
			word.match(quPattern)[0].length;
		return word.slice(splitIndex) + word.slice(0, splitIndex) + 'ay';
	}
	if (word.match(vowelPattern)) {
		return word + 'ay';
	}
	if (word.match(consonantPattern)) {
		let splitIndex = consonantPattern.exec(word).index +
			word.match(consonantPattern)[0].length;
		return word.slice(splitIndex) + word.slice(0, splitIndex) + 'ay';
	}
}


module.exports = {
	translate
}

File: web-development-101/learn-ruby-test-driven/04_pig_latin/pig_latin.rb
#write your code here
def translate(text)
	words = text.split(" ")
	new_words = []
	words.length.times do |i|
		new_words << getPigLatin(words[i])
	end
	new_words.join(" ")
end

def getPigLatin(word)
	vowel_pattern = /^[aeiou]+/i
	consonant_pattern = /^[^aeiou]+/i
    qu_pattern = /[^aeiou]*qu/i
    ret_word = ""
	if word.match(qu_pattern)
		first_index = word =~ qu_pattern
		split_index = first_index + word[qu_pattern].length
		puts first_index
		puts word[qu_pattern]
		puts word[qu_pattern].length
		puts split_index
		ret_word = word[split_index...] + word[0...split_index] + "ay"
	elsif word.match(vowel_pattern)
		ret_word = word + "ay"
	elsif word.match(consonant_pattern)
		first_index = word =~ consonant_pattern
		split_index = first_index + word[consonant_pattern].length
		ret_word = word[split_index...] + word[0...split_index] + "ay"
	end
	ret_word
end

puts translate("quiet")